tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/runtime/api/RuntimeAPITest.java

```diff
@@ -0,0 +1,141 @@
+        ProcessBuilder compile = new ProcessBuilder(compileCmdSet).directory(javaSrcLocation.toFile());
+        try {
+            Process process = compile.start();
+            process.waitFor();
+        } catch (InterruptedException | IOException e) {
+            throw new BallerinaTestException("Error occurred while compiling the java file");
+        }
+
+        List<String> unzipProcessCmdSet = new ArrayList<>();
+        unzipProcessCmdSet.add("unzip");
+        unzipProcessCmdSet.add("-qq");
+        unzipProcessCmdSet.add(jarPath.toString());
+        unzipProcessCmdSet.add("-d");
+        unzipProcessCmdSet.add("targetDir");
+        ProcessBuilder unzipProcess = new ProcessBuilder(unzipProcessCmdSet).directory(javaSrcLocation.toFile());
+        unzipProcess.redirectErrorStream(true);
+        try {
+            Process process = unzipProcess.start();
+            process.waitFor();
+        } catch (IOException | InterruptedException e) {
+            throw new BallerinaTestException("Error occurred while unzipping the jar file");
+        }
+
+        List<String> jarCmdSet = new ArrayList<>();
+        jarCmdSet.add("jar");
+        jarCmdSet.add("-cfe");
+        jarCmdSet.add("test-exec.jar");
+        jarCmdSet.add("org.ballerinalang.test.runtime.api.RuntimeAPICall");
+        jarCmdSet.add(".");
+        Path targetDir = Paths.get(javaSrcLocation.toString(), "targetDir").toAbsolutePath();
+        ProcessBuilder jarProcess = new ProcessBuilder(jarCmdSet).inheritIO().directory(targetDir.toFile());
+        try {
+            Process process = jarProcess.start();
+            process.waitFor();
+        } catch (IOException | InterruptedException e) {
+            throw new BallerinaTestException("Error occurred while packing the jar file");
+        }
+
+        Path execJarPath = Paths.get(javaSrcLocation.toString(), "targetDir", "test-exec.jar").toAbsolutePath();
+        List<String> runCmdSet = new ArrayList<>();
+        runCmdSet.add("java");
+        runCmdSet.add("-jar");
+        runCmdSet.add(execJarPath.toString());
+        ProcessBuilder runProcessBuilder = new ProcessBuilder(runCmdSet);
+        String jacocoArgLine = "-javaagent:" + Paths.get(System.getProperty(BALLERINA_HOME), "bre", "lib",
+                "jacocoagent.jar") + "=destfile=" + Paths.get(System.getProperty(USER_DIR))
+                .resolve("build").resolve("jacoco").resolve("test.exec") + " ";
+        runProcessBuilder.environment().put("JAVA_OPTS", jacocoArgLine);
+        runProcessBuilder.redirectErrorStream(true);
+        Path outputFileDir = Paths.get(javaSrcLocation.toString(), "targetDir", "output.txt");
+        runProcessBuilder.redirectOutput(new File(outputFileDir.toAbsolutePath().toString()));
+        try {
+            Process runProcess = runProcessBuilder.start();
+            runProcess.waitFor();
+            String programOutput = Files.readString(outputFileDir);
```

Comment:
If we refactor the methods, shall we add negative tests for scenarios like starting without init?